In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns

Overview of all the tables

1. ORDER

In [ ]:
orders = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_orders_dataset.csv')
orders.head()

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


In [ ]:
orders.isna().sum()

,0
order_id,0
customer_id,0
order_status,0
order_purchase_timestamp,0
order_approved_at,160
order_delivered_carrier_date,1783
order_delivered_customer_date,2965
order_estimated_delivery_date,0


In [ ]:
orders.duplicated().sum()

np.int64(0)

2. ORDER ITEM

In [ ]:
items = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_order_items_dataset.csv')
items.head()

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


In [ ]:
items.isna().sum()

,0
order_id,0
order_item_id,0
product_id,0
seller_id,0
shipping_limit_date,0
price,0
freight_value,0


In [ ]:
items.duplicated().sum()

np.int64(0)

3. CUSTOMERS

In [ ]:
customers = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_customers_dataset.csv')
customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [ ]:
customers.isna().sum()

,0
customer_id,0
customer_unique_id,0
customer_zip_code_prefix,0
customer_city,0
customer_state,0


In [ ]:
customers.duplicated().sum()

np.int64(0)

4. PRODUCTS

In [ ]:
products = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_products_dataset.csv')
products.head()

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


In [ ]:
products.isna().sum()

,0
product_id,0
product_category_name,610
product_name_lenght,610
product_description_lenght,610
product_photos_qty,610
product_weight_g,2
product_length_cm,2
product_height_cm,2
product_width_cm,2


5. PAYMENT

In [ ]:
payments = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_order_payments_dataset.csv')
payments.head()

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [ ]:
payments.isna().sum()

,0
order_id,0
payment_sequential,0
payment_type,0
payment_installments,0
payment_value,0


In [ ]:
payments.duplicated().sum()

np.int64(0)

6. REVIEWS

In [ ]:
reviews = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_order_reviews_dataset.csv')
reviews.head()

,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18 00:00:00,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10 00:00:00,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17 00:00:00,2018-02-18 14:36:24
3,e64fb393e7b32834bb789ff8bb30750e,658677c97b385a9be170737859d3511b,5,NaN,Recebi bem antes do prazo estipulado.,2017-04-21 00:00:00,2017-04-21 22:02:06
4,f7c4243c7fe1938f181bec41a392bdeb,8e6bfb81e283fa7e4f11123a3fb894f1,5,NaN,Parabéns lojas lannister adorei comprar pela I...,2018-03-01 00:00:00,2018-03-02 10:26:53


In [ ]:
reviews.isna().sum()

,0
review_id,0
order_id,0
review_score,0
review_comment_title,87656
review_comment_message,58247
review_creation_date,0
review_answer_timestamp,0


In [ ]:
reviews.duplicated().sum()

np.int64(0)

6. SELLERS

In [ ]:
sellers = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_sellers_dataset.csv')
sellers.head()

,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ
3,c0f3eea2e14555b6faeea3dd58c1b1c3,4195,sao paulo,SP
4,51a04a8a6bdcb23deccc82b0b80742cf,12914,braganca paulista,SP


In [ ]:
sellers.isna().sum()

,0
seller_id,0
seller_zip_code_prefix,0
seller_city,0
seller_state,0


In [ ]:
sellers.duplicated().sum()

np.int64(0)

7. TRANSLATION

In [ ]:
translation = pd.read_csv('/content/drive/MyDrive/Ecommerce/product_category_name_translation.csv')
translation.head()

,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto
3,cama_mesa_banho,bed_bath_table
4,moveis_decoracao,furniture_decor


In [ ]:
translation.isna().sum()

,0
product_category_name,0
product_category_name_english,0


8. GEOLOCATION

In [ ]:
geolocation = pd.read_csv('/content/drive/MyDrive/Ecommerce/olist_geolocation_dataset.csv')
geolocation.head()

,geolocation_zip_code_prefix,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,1037,-23.545621,-46.639292,sao paulo,SP
1,1046,-23.546081,-46.644820,sao paulo,SP
2,1046,-23.546129,-46.642951,sao paulo,SP
3,1041,-23.544392,-46.639499,sao paulo,SP
4,1035,-23.541578,-46.641607,sao paulo,SP


In [ ]:
geolocation.duplicated().sum()

np.int64(261831)

In [ ]:
geolocation = geolocation.drop_duplicates(subset=['geolocation_zip_code_prefix'], keep='first')

In [ ]:
geolocation.duplicated().sum()

np.int64(0)

Merge the table in one big table

In [ ]:
products = pd.merge(products, translation, on='product_category_name', how='left')
products

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm,product_category_name_english
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0,perfumery
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0,art
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0,sports_leisure
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0,baby
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0,housewares
...,...,...,...,...,...,...,...,...,...,...
32946,a0b7d5a992ccda646f2d34e418fff5a0,moveis_decoracao,45.0,67.0,2.0,12300.0,40.0,40.0,40.0,furniture_decor
32947,bf4538d88321d0fd4412a93c974510e6,construcao_ferramentas_iluminacao,41.0,971.0,1.0,1700.0,16.0,19.0,16.0,construction_tools_lights
32948,9a7c6041fa9592d9d9ef6cfe62a71f8c,cama_mesa_banho,50.0,799.0,1.0,1400.0,27.0,7.0,27.0,bed_bath_table
32949,83808703fc0706a22e264b9d75f04a2e,informatica_acessorios,60.0,156.0,2.0,700.0,31.0,13.0,20.0,computers_accessories


In [ ]:
products.drop(columns=['product_category_name'], inplace=True)
products.rename(columns={'product_category_name_english': 'product_category'}, inplace=True)

In [ ]:
df = pd.merge(orders, customers, on='customer_id', how='left')
df = pd.merge(df, items, on='order_id', how='left')
df = pd.merge(df, payments, on='order_id', how='left')
df = pd.merge(df, reviews, on='order_id', how='left')
df = pd.merge(df, sellers, on='seller_id', how='left')
df = pd.merge(df, products, on='product_id', how='left')

In [ ]:
df

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,customer_unique_id,customer_zip_code_prefix,...,seller_city,seller_state,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm,product_category
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,7c396fd4830fd04220f754e42b4e5bff,3149,...,maua,SP,40.0,268.0,4.0,500.0,19.0,8.0,13.0,housewares
1,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,7c396fd4830fd04220f754e42b4e5bff,3149,...,maua,SP,40.0,268.0,4.0,500.0,19.0,8.0,13.0,housewares
2,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,7c396fd4830fd04220f754e42b4e5bff,3149,...,maua,SP,40.0,268.0,4.0,500.0,19.0,8.0,13.0,housewares
3,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00,af07308b275d755c9edb36a90c618231,47813,...,belo horizonte,SP,29.0,178.0,1.0,400.0,19.0,13.0,19.0,perfumery
4,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00,3a653a41f6f9fc3d2a113cf8398680e8,75265,...,guariba,SP,46.0,232.0,1.0,420.0,24.0,19.0,21.0,auto
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
119138,63943bddc261676b46f01ca7ac2f7bd8,1fca14ff2861355f6e5f14306ff977a7,delivered,2018-02-06 12:58:58,2018-02-06 13:10:37,2018-02-07 23:22:42,2018-02-28 17:37:56,2018-03-02 00:00:00,da62f9e57a76d978d02ab5362c509660,11722,...,tupa,SP,52.0,828.0,4.0,4950.0,40.0,10.0,40.0,baby
119139,83c1379a015df1e13d02aae0204711ab,1aa71eb042121263aafbe80c1b562c9c,delivered,2017-08-27 14:46:43,2017-08-27 15:04:16,2017-08-28 20:52:26,2017-09-21 11:24:17,2017-09-27 00:00:00,737520a9aad80b3fbbdad19b66b37b30,45920,...,sao paulo,SP,51.0,500.0,2.0,13300.0,32.0,90.0,22.0,home_appliances_2
119140,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,5097a5312c8b157bb7be58ae360ef43c,28685,...,ilicinea,MG,59.0,1893.0,1.0,6550.0,20.0,20.0,20.0,computers_accessories
119141,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,5097a5312c8b157bb7be58ae360ef43c,28685,...,ilicinea,MG,59.0,1893.0,1.0,6550.0,20.0,20.0,20.0,computers_accessories


In [ ]:
df = pd.merge(df, geolocation, left_on='customer_zip_code_prefix', right_on='geolocation_zip_code_prefix', how='left')

In [ ]:
df.drop(columns=['geolocation_zip_code_prefix'], inplace=True)

In [ ]:
columns_to_drop = [
    'review_id',
    'review_comment_title',
    'review_comment_message',
    'customer_unique_id',
    'customer_zip_code_prefix',
    'seller_zip_code_prefix',
    'order_item_id'
]
df.drop(columns=columns_to_drop, inplace=True, errors='ignore')

In [ ]:
df

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,customer_city,customer_state,...,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm,product_category,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,500.0,19.0,8.0,13.0,housewares,-23.574809,-46.587471,sao paulo,SP
1,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,500.0,19.0,8.0,13.0,housewares,-23.574809,-46.587471,sao paulo,SP
2,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,500.0,19.0,8.0,13.0,housewares,-23.574809,-46.587471,sao paulo,SP
3,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00,barreiras,BA,...,1.0,400.0,19.0,13.0,19.0,perfumery,-12.169860,-44.988369,barreiras,BA
4,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00,vianopolis,GO,...,1.0,420.0,24.0,19.0,21.0,auto,-16.746337,-48.514624,vianopolis,GO
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
119138,63943bddc261676b46f01ca7ac2f7bd8,1fca14ff2861355f6e5f14306ff977a7,delivered,2018-02-06 12:58:58,2018-02-06 13:10:37,2018-02-07 23:22:42,2018-02-28 17:37:56,2018-03-02 00:00:00,praia grande,SP,...,4.0,4950.0,40.0,10.0,40.0,baby,-24.001467,-46.446355,praia grande,SP
119139,83c1379a015df1e13d02aae0204711ab,1aa71eb042121263aafbe80c1b562c9c,delivered,2017-08-27 14:46:43,2017-08-27 15:04:16,2017-08-28 20:52:26,2017-09-21 11:24:17,2017-09-27 00:00:00,nova vicosa,BA,...,2.0,13300.0,32.0,90.0,22.0,home_appliances_2,-17.891522,-39.370942,nova vicosa,BA
119140,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,japuiba,RJ,...,1.0,6550.0,20.0,20.0,20.0,computers_accessories,-22.555985,-42.690761,japuiba,RJ
119141,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,japuiba,RJ,...,1.0,6550.0,20.0,20.0,20.0,computers_accessories,-22.555985,-42.690761,japuiba,RJ


In [ ]:
extra_columns_to_drop = [
    'product_name_lenght',
    'product_weight_g',
    'product_length_cm',
    'product_height_cm',
    'product_width_cm',
    'geolocation_city',
    'geolocation_state',
    'payment_sequential'
]

df.drop(columns=extra_columns_to_drop, inplace=True, errors='ignore')

In [ ]:
df

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,customer_city,customer_state,...,review_score,review_creation_date,review_answer_timestamp,seller_city,seller_state,product_description_lenght,product_photos_qty,product_category,geolocation_lat,geolocation_lng
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,2017-10-11 00:00:00,2017-10-12 03:43:48,maua,SP,268.0,4.0,housewares,-23.574809,-46.587471
1,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,2017-10-11 00:00:00,2017-10-12 03:43:48,maua,SP,268.0,4.0,housewares,-23.574809,-46.587471
2,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,sao paulo,SP,...,4.0,2017-10-11 00:00:00,2017-10-12 03:43:48,maua,SP,268.0,4.0,housewares,-23.574809,-46.587471
3,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00,barreiras,BA,...,4.0,2018-08-08 00:00:00,2018-08-08 18:37:50,belo horizonte,SP,178.0,1.0,perfumery,-12.169860,-44.988369
4,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00,vianopolis,GO,...,5.0,2018-08-18 00:00:00,2018-08-22 19:07:58,guariba,SP,232.0,1.0,auto,-16.746337,-48.514624
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
119138,63943bddc261676b46f01ca7ac2f7bd8,1fca14ff2861355f6e5f14306ff977a7,delivered,2018-02-06 12:58:58,2018-02-06 13:10:37,2018-02-07 23:22:42,2018-02-28 17:37:56,2018-03-02 00:00:00,praia grande,SP,...,4.0,2018-03-01 00:00:00,2018-03-02 17:50:01,tupa,SP,828.0,4.0,baby,-24.001467,-46.446355
119139,83c1379a015df1e13d02aae0204711ab,1aa71eb042121263aafbe80c1b562c9c,delivered,2017-08-27 14:46:43,2017-08-27 15:04:16,2017-08-28 20:52:26,2017-09-21 11:24:17,2017-09-27 00:00:00,nova vicosa,BA,...,5.0,2017-09-22 00:00:00,2017-09-22 23:10:57,sao paulo,SP,500.0,2.0,home_appliances_2,-17.891522,-39.370942
119140,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,japuiba,RJ,...,2.0,2018-01-26 00:00:00,2018-01-27 09:16:56,ilicinea,MG,1893.0,1.0,computers_accessories,-22.555985,-42.690761
119141,11c177c8e97725db2631073c19f07b62,b331b74b18dc79bcdf6532d51e1637c1,delivered,2018-01-08 21:28:27,2018-01-08 21:36:21,2018-01-12 15:35:03,2018-01-25 23:32:54,2018-02-15 00:00:00,japuiba,RJ,...,2.0,2018-01-26 00:00:00,2018-01-27 09:16:56,ilicinea,MG,1893.0,1.0,computers_accessories,-22.555985,-42.690761


In [ ]:
df.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date',
       'customer_city', 'customer_state', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value', 'payment_type',
       'payment_installments', 'payment_value', 'review_score',
       'review_creation_date', 'review_answer_timestamp', 'seller_city',
       'seller_state', 'product_description_lenght', 'product_photos_qty',
       'product_category', 'geolocation_lat', 'geolocation_lng'],
      dtype='object')

Overview

In [ ]:
df.shape

(119143, 28)

In [ ]:
df.duplicated().sum()

np.int64(11303)

In [ ]:
df.drop_duplicates(inplace=True)

In [ ]:
df.duplicated().sum()

np.int64(0)

In [ ]:
df.isna().sum()

,0
order_id,0
customer_id,0
order_status,0
order_purchase_timestamp,0
order_approved_at,166
order_delivered_carrier_date,1892
order_delivered_customer_date,3163
order_estimated_delivery_date,0
customer_city,0
customer_state,0


In [ ]:
(df.isna().sum())/len(df)*100

,0
order_id,0.000000
customer_id,0.000000
order_status,0.000000
order_purchase_timestamp,0.000000
order_approved_at,0.153932
order_delivered_carrier_date,1.754451
order_delivered_customer_date,2.933049
order_estimated_delivery_date,0.000000
customer_city,0.000000
customer_state,0.000000


In [ ]:
df['product_category'] = df['product_category'].fillna('Unknown')
df['product_description_lenght'] = df['product_description_lenght'].fillna(0)

In [ ]:
df['price'] = df['price'].fillna(0)
df['freight_value'] = df['freight_value'].fillna(0)
df['payment_value'] = df['payment_value'].fillna(0)

In [ ]:
df['payment_installments'] = df['payment_installments'].fillna(1)

In [ ]:
df['review_creation_date'] = df['review_creation_date'].fillna('No Review Sent')
df['review_answer_timestamp'] = df['review_answer_timestamp'].fillna('No Answer')

In [ ]:
df['review_score'] = df['review_score'].fillna(-1)

In [ ]:
df['payment_installments'] = df['payment_installments'].fillna(1)

In [ ]:
(df.isna().sum())/len(df)*100

,0
order_id,0.000000
customer_id,0.000000
order_status,0.000000
order_purchase_timestamp,0.000000
order_approved_at,0.153932
order_delivered_carrier_date,1.754451
order_delivered_customer_date,2.933049
order_estimated_delivery_date,0.000000
customer_city,0.000000
customer_state,0.000000


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 107840 entries, 0 to 119142
Data columns (total 28 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       107840 non-null  object 
 1   customer_id                    107840 non-null  object 
 2   order_status                   107840 non-null  object 
 3   order_purchase_timestamp       107840 non-null  object 
 4   order_approved_at              107674 non-null  object 
 5   order_delivered_carrier_date   105948 non-null  object 
 6   order_delivered_customer_date  104677 non-null  object 
 7   order_estimated_delivery_date  107840 non-null  object 
 8   customer_city                  107840 non-null  object 
 9   customer_state                 107840 non-null  object 
 10  product_id                     107026 non-null  object 
 11  seller_id                      107026 non-null  object 
 12  shipping_limit_date            1070

In [ ]:
date_columns=['order_purchase_timestamp',
              'order_approved_at',
              'order_delivered_carrier_date',
              'order_delivered_customer_date',
              'order_estimated_delivery_date',
              'shipping_limit_date',
              'review_answer_timestamp',
              'review_creation_date']

for i in date_columns:
  df[i]=pd.to_datetime(df[i],errors='coerce')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 107840 entries, 0 to 119142
Data columns (total 28 columns):
 #   Column                         Non-Null Count   Dtype         
---  ------                         --------------   -----         
 0   order_id                       107840 non-null  object        
 1   customer_id                    107840 non-null  object        
 2   order_status                   107840 non-null  object        
 3   order_purchase_timestamp       107840 non-null  datetime64[ns]
 4   order_approved_at              107674 non-null  datetime64[ns]
 5   order_delivered_carrier_date   105948 non-null  datetime64[ns]
 6   order_delivered_customer_date  104677 non-null  datetime64[ns]
 7   order_estimated_delivery_date  107840 non-null  datetime64[ns]
 8   customer_city                  107840 non-null  object        
 9   customer_state                 107840 non-null  object        
 10  product_id                     107026 non-null  object        
 11  selle

In [ ]:
col=['payment_type','order_status','payment_installments']

for i in col:
  print(f'{df[i].value_counts()}\n')

payment_type
credit_card    80218
boleto         20597
voucher         5422
debit_card      1599
not_defined        3
Name: count, dtype: int64

order_status
delivered      104678
shipped          1192
canceled          662
unavailable       643
invoiced          339
processing        319
created             5
approved            2
Name: count, dtype: int64

payment_installments
1.0     53751
2.0     12818
3.0     10908
4.0      7441
10.0     5871
5.0      5520
8.0      4593
6.0      4160
7.0      1711
9.0       681
12.0      153
15.0       81
18.0       31
11.0       25
24.0       22
20.0       19
13.0       18
14.0       15
17.0        8
16.0        6
21.0        4
0.0         2
23.0        1
22.0        1
Name: count, dtype: int64



**OverView on Categorical & Numerical Data**

In [ ]:
df.describe(include='O')

,order_id,customer_id,order_status,customer_city,customer_state,product_id,seller_id,payment_type,seller_city,seller_state,product_category
count,107840,107840,107840,107840,107840,107026,107026,107839,107026,107026,107840
unique,99441,99441,8,4119,27,32951,3095,5,611,23,72
top,fa65dad1b0e818e3ccc5cb0e39231352,9af2372a1e49340278e7c1ef8d749f34,delivered,sao paulo,SP,99a4788cb24856965c36a24e339b6058,6560211a19b47992c3666cc44a7e94c0,credit_card,sao paulo,SP,bed_bath_table
freq,25,25,104678,16930,45342,504,2064,80218,26309,76111,10874


1. The data has 99,441 unique orders and 99,441 unique customer.
2. Most of the orders are successfully delivered.
3. sao paulo is the top city for both customers and sellers. This makes perfect sense, as more local sellers mean faster and cheaper shipping for local customers.
4. There are 72 different product categories available.
5. Credit Card is the most popular and frequent payment type used.